homeassistant/components/simplisafe/__init__.py

```diff
@@ -474,6 +475,27 @@ async def async_init(self):
                 _LOGGER.error("Error while fetching initial event: %s", err)
                 self.initial_event_to_use[system.system_id] = {}
 
+            # The initial update doesn't process notifications because it is likely that
+            # they will trigger SIMPLISAFE_NOTIFICATION events before other critical
+            # components – such as automation – are ready. Therefore, we schedule an
+            # update of notifications as soon as HASS is fully started:
+            async def process_notifications_listener(_):
+                """Define a listener to run when `homeassistant_start` fires."""
+
+                async def process_when_running():
+                    """Process notifications when HASS is running."""
+                    while self._hass.state != CoreState.running:
+                        await asyncio.sleep(1)
+                    self._async_process_new_notifications(  # pylint: disable=W0640
+                        system
+                    )
+
+                return asyncio.create_task(process_when_running())
```

Comment:
Are we using the return value somewhere?

I still think we're should make sure to clean up this task on home assistant stop if it's stale.

Reply:
We can do that by changing the `while` to be while `self._hass.state == CoreState.starting`.

However, we should still catch if Simplisafe is initialized after Home Assistant has started (when config entry created).

Reply:
I simplified things (via 86af) by allowing the integration to init no matter what, then doing this "has HASS started?" check when it actually goes to process notifications (which it does after setup is complete). As before, this process won't start if the integration fails setup. Thoughts?